x/investors/keeper/genesis.go

```diff
@@ -0,0 +1,63 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/milkyway-labs/milkyway/v10/x/investors/types"
+)
+
+// ExportGenesis returns the GenesisState associated with the given context
+func (k *Keeper) ExportGenesis(ctx sdk.Context) *types.GenesisState {
+	investorsRewardRatio, err := k.GetInvestorsRewardRatio(ctx)
+	if err != nil {
+		panic(err)
+	}
+
+	vestingInvestorsAddrs, err := k.GetAllVestingInvestorsAddresses(ctx)
+	if err != nil {
+		panic(err)
+	}
+
+	return types.NewGenesisState(investorsRewardRatio, vestingInvestorsAddrs)
+}
+
+// InitGenesis initializes the state from a GenesisState
+func (k *Keeper) InitGenesis(ctx sdk.Context, state *types.GenesisState) error {
+	err := k.SetInvestorsRewardRatio(ctx, state.InvestorsRewardRatio)
+	if err != nil {
+		return err
+	}
+
+	for _, investor := range state.VestingInvestorsAddresses {
+		err = k.SetVestingInvestor(ctx, investor)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Initialize lookup table for delegator addresses. Note that this should be done
+	// manually in the investors module, because the distribution module's
+	// AfterSetWithdrawAddress hook is not called when initializing the distribution
+	// module from genesis state.
+	k.distrKeeper.IterateDelegatorWithdrawAddrs(ctx, func(del, addr sdk.AccAddress) (stop bool) {
```

Comment:
Should we do this in the migration handler to ensure that the revese lookup map have the correct associations?